### ISSUE TITLE
Decide what to do about Advisor Tutorials / ad-hoc messages

### ISSUE THREAD

**Issue by aubergine10** [CitiesSkylinesMods/TMPE#1240]
As mentioned in https://github.com/CitiesSkylinesMods/TMPE/issues/1239#issuecomment-997451706 :

> it works if you turn on guide. after we wrote this code CO made a change that either you will have all guides+tutorials every time you start a new city or you will have no guide at all. ppl will go with the latter option rendering the guide system useless.
> 
> This also influences TMPE tutorials. we should have a system to override the guide system or just abandon it and write our own framework.

To clarify, TM:PE **advisor guide panels** will only work when both vanilla and TM:PE settings are **enabled**:

* System Settings > Gameplay > Show in-game guide popups
* TM:PE settings > General > Enable tutorial messages

There are currently two types of "tutorial messages" that I'm aware of in TM:PE:

* The big "Advisor Guide" popups that appear when:
    * TM:PE toolbar is opened first time
    * A tool is used for first time
    * Green (?) button near vanilla roads menu is clicked while TM:PE open
* Ad-hoc notifications to user while using a tool:
    * Unable to use lane arrows (see #1239)
    * Are there more?

IMO:

1. "Advisor Guides" should stay, and should be controlled entirely by the vanilla **Show in-game guide popups** option.
    * Some other popular mods use those guides, eg. Landscaping Tools, Trailer Variation Loader, etc.
    * That sort of guide is familiar to users and provides sufficient text for summarising what a tool does
2. TM:PE **Enable tutorial messages** option should be removed
    * It's entirely dependent on the vanilla setting, we might as well just use vanilla setting
3. Ad-hoc warnings should be implemented by some other method (will be specific to each use case IMO)

Additionally, I found one other issue with the advisor text guides - the URL is to the old documentation site!

![image](https://user-images.githubusercontent.com/1386719/146691363-1b614bc2-0991-4eb1-9536-04a097405747.png)

For now I think we should change that URL to just point at either http://tmpe.me or https://github.com/CitiesSkylinesMods/TMPE/wiki - we should also consider putting something more useful than just a URL in there.

**Comment by krzychu124** [comment]
Additionally, maybe we should explore build-in tutorial panel and inject some TM:PE tutorials(that one you can open from the in-game options menu)?

**Comment by aubergine10** [comment]
> Additionally, maybe we should explore build-in tutorial panel and inject some TM:PE tutorials(that one you can open from the in-game options menu)?

I was today years old when I first opened the tutorial log from in-game pause menu. :)

Yup, would be nice, but preferably add a `Mods` tab (so other mods can use it too if desired) to separate out from the vanilla clutter.

**Comment by kvakvs** [comment]
I personally never found value in any advisor tutorials, they're just annoying popups I (and many other people) click away before even focusing on their first word. There must be a better way, even a link to web page is better than a huge white box blocking quarter of screen.

**Comment by aubergine10** [comment]
Yup, I usually ignore them. I remember when game first came out and it was spamming advisor popups at the start of every game.

----

For **Lane Arrows** tool, UI could show what's preventing use of lane arrows.

Example: If lane customised with lane connectors, show just one button with the lane connector icon on it (same icon from the main toolbar button). Button tooltip = the text we used to show in advisor hint popup.

That way user sees at glance that they can't customise that lane, and why. Newbies can hover button to learn more.

**Comment by kvakvs** [comment]
I would say, if we draw a round button from the UI on the world, it should be clickable and switch to edit mode in lane arrows!
Same goes for timed TL, if we prevent some edits on a node because its in TTL, the icon over the node would be clickable and would open TTL
